src/main/java/edu/ucdavis/dss/ipa/entities/SectionGroupCost.java

```diff
@@ -18,6 +19,7 @@
 @Table(name = "SectionGroupCosts")
 @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
 @JsonDeserialize(using = SectionGroupCostDeserializer.class)
+@ValidSectionGroupCost
```

Comment:
I love the use of validation but this custom validator at the entity level I think is a misused pattern. Spring has at least two ways to write custom validators. The way you implemented `@ValidSectionGroupCost` is really for creating entirely new _member_ level validators that you could sprinkle across multiple entities.

If you wanted to instead validate `SectionGroupCost` itself, a class-level validator like the one in `CourseValidator.java` is cleaner I think. It requires fewer files and does not create a custom annotation which implies it should be used more than once.